test/powershell/Modules/Microsoft.PowerShell.Utility/New-Guid.Tests.ps1

```diff
@@ -2,17 +2,38 @@
 # Licensed under the MIT License.
 Describe "New-Guid" -Tags "CI" {
 
-    It "returns a new guid" {
+    It "Returns a new guid" {
         $guid = New-Guid
         $guid | Should -BeOfType System.Guid
     }
 
-    It "should not be all zeros" {
+    It "Should not be all zeros" {
         $guid = New-Guid
         $guid.ToString() | Should -Not -BeExactly "00000000-0000-0000-0000-000000000000"
     }
 
-    It "should return different guids with each call" {
+    It "Should be all zeros" {
+        $guid = New-Guid -Empty
+        $guid.ToString() | Should -BeExactly "00000000-0000-0000-0000-000000000000"
+    }
+
+    It "Should convert a string to a guid" {
+        $guid1 = New-Guid
+        $guid2 = New-Guid -InputObject $guid1.ToString()
+        $guid3 = New-Guid $guid1.ToString()
+        $guid2 | Should -BeOfType System.Guid
+        $guid1.ToString() | Should -BeExactly $guid2.ToString()
+        $guid1.ToString() | Should -BeExactly $guid3.ToString()
+    }
+
+    It "Should convert a string to a guid, value from pipeline" {
+        $guid1 = New-Guid
+        $guid2 = $guid1.ToString() | New-Guid
+        $guid2 | Should -BeOfType System.Guid
+        $guid1.ToString() | Should -BeExactly $guid2.ToString()
+    }
+
+    It "Should return different guids with each call" {
```

Comment:
I think we need a test that uses pipeline input for multiple values in the pipeline.
Something like:

```
$guids = '11c43ee8-b9d3-4e51-b73f-bd9dda66e29c','0f8fad5bd9cb469fa16570867728950e' | New-Guid
$guids[0].ToString() | Should -BeExactly '11c43ee8-b9d3-4e51-b73f-bd9dda66e29c'
$guids[1].ToString() | Should -BeExactly '0f8fad5b-d9cb-469f-a165-70867728950e'
```

This tests several things
- Conversion of guids in different formats
- Conversion of guids not created by New-Guid
- Conversion of multiple values from the pipeline